### PR TITLE
Integrate test manager into RPClient

### PR DESCRIPTION
**Pull request by pahniki**

### CODE REVIEW

File: reportportal_client/client.pyi [hunk 1]
```diff
@@ -1,57 +1,26 @@
...
+    token: Text = ...
+    launch_id: Optional[Text] = ...
+    verify_ssl: bool = ...
+    is_skipped_an_issue: bool = ...
+    api_v1: str = ...
+    api_v2: str = ...
+    base_url_v1: Text = ...
+    base_url_v2: Text = ...
+    session: Session = ...
+    def __init__(self, endpoint: str, project: str, token: str, log_batch_size: int=..., is_skipped_an_issue: bool=..., verify_ssl: bool=..., retries: Optional[int]=..., max_pool_size: int=..., launch_id: Optional[str]=...) -> None: ...
```
Comment (iivanou) [inline reportportal_client/client.pyi:21]: Please, use indentation in .pyi files as well.

File: reportportal_client/core/test_manager.py [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+            {
+                "<key1>": "<value1>",
+                "<key2>": "<value2>",
+                ...
+            }
+    """
+
+    def __init__(self,
+                 session,  # type: requests.Session
+                 endpoint,
```
Comment (iivanou) [inline reportportal_client/core/test_manager.py:48]: type: str

File: reportportal_client/core/test_manager.py [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+        :param api_version:     RP API version
+        :param time:            log time
+        :param message:         log message
+        :param level:           log level
+        :param attachment:      attachments to log (images,files,etc.)
+        :param item_id:         parent item UUID
+        :return:                log item UUID
+        """
+        uuid = generate_uuid()
+        # Todo: Do we store log items?
```
Comment (iivanou) [inline reportportal_client/core/test_manager.py:192]: We do not store them.

**Review by iivanou — COMMENTED** [review]
Rebase please and fix conflicts.

File: reportportal_client/client.pyi (hunk 1 above)
Comment (pahniki) [reply reportportal_client/client.pyi]: Done

File: reportportal_client/core/test_manager.py (hunk 2 above)
Comment (pahniki) [reply reportportal_client/core/test_manager.py]: Done

File: reportportal_client/core/test_manager.py (hunk 3 above)
Comment (pahniki) [reply reportportal_client/core/test_manager.py]: Done

File: reportportal_client/client.py [hunk 4]
```diff
@@ -14,87 +14,214 @@
...
+            "description": description,
+            "attributes": attributes,
+            "startTime": start_time,
+            "mode": mode,
+            "rerun": rerun,
+            "rerunOf": rerun_of
+        }
+        data.update(kwargs)
+        url = uri_join(self.base_url_v2, "launch")
+        response = self.session.post(url=url, json=data,
```
Comment (iivanou) [inline reportportal_client/client.py:122]: Let's use our new Request objects instead.

File: reportportal_client/client.py [hunk 5]
```diff
@@ -14,87 +14,214 @@
...
+        if attributes and isinstance(attributes, dict):
+            attributes = dict_to_payload(attributes)
+        data = {
+            "endTime": end_time,
+            "status": status,
+            "attributes": attributes
+        }
+        data.update(kwargs)
+        url = uri_join(self.base_url_v2, "launch", self.launch_id, "finish")
+        response = self.session.put(url=url, json=data, verify=self.verify_ssl)
```
Comment (iivanou) [inline reportportal_client/client.py:154]: Let's use our new Request objects instead.

File: reportportal_client/client.py [hunk 6]
```diff
@@ -14,87 +14,214 @@
...
+        self.log_batch_size = log_batch_size  # type: int
+        self.project = project  # type: str
+        self.token = token  # type: str
+        self.launch_id = launch_id  # type: str
+        self.verify_ssl = verify_ssl  # type: bool
+        self.is_skipped_an_issue = is_skipped_an_issue  # type: bool
+
+        self.api_v1 = 'v1'
+        self.api_v2 = 'v2'
+        self.base_url_v1 = uri_join(self.endpoint, f"api/{self.api_v1}",
```
Comment (iivanou) [inline reportportal_client/client.py:70]: It's not Python 2 compatible.

File: reportportal_client/client.pyi [hunk 7]
```diff
@@ -1,57 +1,43 @@
...
-            self,
-            launch_uuid: Text,
-            end_time: Text,
-            **kwargs: Any) -> None: ...
-    def save_log(
-            self,
-            launch_uuid: Text,
-            log_time: Text,
-            **kwargs: Any) -> None: ...
+    endpoint: str = ...
```
Comment (iivanou) [inline reportportal_client/client.pyi:5]: For Python two it should be Text type.

File: reportportal_client/helpers.pyi [hunk 8]
```diff
@@ -1,21 +1,24 @@
...
-def verify_value_length(
-        attributes: Union[List[Dict], None]) -> Union[List[Dict], None]: ...
+def generate_uuid() -> str: ...
+def convert_string(value: str) -> str: ...
+def dict_to_payload(dictionary: dict) -> list[dict]: ...
+def gen_attributes(rp_attributes: list) -> list[dict]: ...
+def get_launch_sys_attrs()-> dict[str]: ...
+def get_package_version(package_name:str) -> str: ...
+def uri_join(*uri_parts: str) -> str: ...
+def get_id(response: Response) -> str: ...
```
Comment (iivanou) [inline reportportal_client/helpers.pyi:17]: Replace str with Text in all *pyi files. P.S. from typing import Text

**Review by iivanou — CHANGES_REQUESTED** [review]
Why did you removed reportportal_client/core/test_manager.pyi?

**Review by iivanou — DISMISSED** [review]

**Pull request merged**